mobile/src/main/java/org/openhab/habdroid/ui/preference/CustomInputTypePreference.kt

```diff
@@ -67,26 +69,52 @@ open class CustomInputTypePreference constructor(context: Context, attrs: Attrib
         return PrefFragment.newInstance(key, title, inputType, autofillHints)
     }
 
-    class PrefFragment : EditTextPreferenceDialogFragmentCompat() {
+    class PrefFragment : EditTextPreferenceDialogFragmentCompat(), TextWatcher {
+        private lateinit var wrapper: TextInputLayout
+        private lateinit var editor: EditText
+
         override fun onBindDialogView(view: View?) {
-            val wrapper = view?.findViewById<TextInputLayout>(R.id.input_wrapper)
-            val editor = view?.findViewById<EditText>(android.R.id.edit)
-            arguments?.getInt(KEY_INPUT_TYPE)?.let { type ->
-                editor?.inputType = type
-            }
-            arguments?.getCharSequence(KEY_TITLE)?.let { title ->
-                wrapper?.hint = title
-            }
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                val hints = arguments?.getStringArray(KEY_AUTOFILL_HINTS)
-                if (hints == null) {
-                    editor?.importantForAutofill = View.IMPORTANT_FOR_AUTOFILL_NO
-                } else {
-                    editor?.importantForAutofill = View.IMPORTANT_FOR_AUTOFILL_YES
-                    editor?.setAutofillHints(*hints)
+            super.onBindDialogView(view)
+            if (view != null) {
+                wrapper = view.findViewById(R.id.input_wrapper)
+                editor = view.findViewById(android.R.id.edit)
+                editor.addTextChangedListener(this)
+                arguments?.getInt(KEY_INPUT_TYPE)?.let { type ->
+                    editor.inputType = type
+                }
+                arguments?.getCharSequence(KEY_TITLE)?.let { title ->
+                    wrapper.hint = title
+                }
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    val hints = arguments?.getStringArray(KEY_AUTOFILL_HINTS)
+                    if (hints == null) {
+                        editor.importantForAutofill = View.IMPORTANT_FOR_AUTOFILL_NO
+                    } else {
+                        editor.importantForAutofill = View.IMPORTANT_FOR_AUTOFILL_YES
+                        editor.setAutofillHints(*hints)
+                    }
                 }
             }
-            super.onBindDialogView(view)
+        }
+
+        override fun beforeTextChanged(charSequence: CharSequence, start: Int, before: Int, count: Int) {
+            // no-op
+        }
+
+        override fun onTextChanged(charSequence: CharSequence, start: Int, before: Int, count: Int) {
+            // no-op
+        }
+
+        override fun afterTextChanged(editable: Editable) {
+            val value = editable.toString()
+
+            val whitespaceAtLastCharacter = value.lastOrNull()?.isWhitespace() ?: false
+
+            val res = editor.resources
+            wrapper.error = when {
+                whitespaceAtLastCharacter -> res.getString(R.string.error_trailing_whitespace)
```

Comment:
I'd add a check for the first character as well.

Reply:
Ok, i added : 

`val isFirstCharWhitespace = value.firstOrNull()?.isWhitespace() ?: false`

to the check :

wrapper?.error = when {
                isFirstCharWhitespace -> res.getString(R.string.error_first_char_is_whitespace)
                isLastCharWhitespace -> res.getString(R.string.error_last_char_is_whitespace)
                else -> null
            }